[mlir][ArmSME] Update docs (#74527)
llvm/llvm-project: mlir/docs/Dialects/ArmSME.md

# 'ArmSME' Dialect

Basic dialect to target Arm SME.

This dialect defines custom and LLVM IR intrinsic operations that are used to

> target Arm Scalable Matrix Extension. Through the available lowerings one can,

Reviewer: Feel free to ignore, I know it's correct, but saying 'one' always always sounds odd to me :P

```suggestion
target Arm Scalable Matrix Extension. Through the available lowerings you can,
```

> target Arm Scalable Matrix Extension. Through the available lowerings one can,

Reviewer: nit: Maybe just say passes? I'm not sure if `-allocate-arm-sme-tiles` is a lowering, but it is used to target SME
```suggestion
target Arm Scalable Matrix Extension. Through the available conversion and ArmSME passes one can,
```

for example, lower a [linalg.matmul](https://mlir.llvm.org/docs/Dialects/Linalg/#linalgmatmul-linalgmatmulop)
opereation to Arm SME
[FMOPA](https://developer.arm.com/documentation/ddi0602/2023-03/SME-Instructions/FMOPA--widening---Half-precision-floating-point-sum-of-outer-products-and-accumulate-)

> (floating point outer product) operations. See one of the in-tree end-to-end

Reviewer: nit:
```suggestion
(floating-point outer product) operations. See one of the in-tree end-to-end
```
